src/global_mapper_ros/global_mapper_ros.cc

```diff
@@ -87,9 +116,11 @@ void GlobalMapperRos::GrayscaleToRGBJet(double v, double vmin, double vmax, std:
   }
 }
 
-void GlobalMapperRos::PublishMap(const ros::TimerEvent& event) {
-  // lock
-  std::unique_lock<std::mutex> map_lock = global_mapper_ptr_->MapLock();
+void GlobalMapperRos::PopulateVoxelMapMsg(visualization_msgs::MarkerArray* marker_array) {
+  // check for bad input
+  if (marker_array == nullptr) {
+    return;
+  }
```

Comment:
Is the raw output really going to be a `MarkerArray`? I think you need two publish calls: one to publish the data and one to publish the visualization message.